test/services/transfers_repository.js

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright: Ambrosus Inc.
+Email: tech@ambrosus.com
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+import chai from 'chai';
+import sinonChai from 'sinon-chai';
+import chaiAsPromised from 'chai-as-promised';
+import TransfersRepository from '../../src/services/transfers_repository';
+import sinon from 'sinon';
+
+chai.use(sinonChai);
+chai.use(chaiAsPromised);
+const {expect} = chai;
+
+describe('Transfers repository', () => {
+  let transferWrapperMock;
+  let configWrapperMock;
+  let activeTransfersCacheMock;
+  let blockchainStateWrapperMock;
+  let transfersEventEmitterWrapper;
+  let transfersRepository;
+
+  describe('prepareResolutionEvent', () => {
+    const donorId = 1;
+    const bundleId = 2;
+    const transferId = 3;
+    const events = [
+      {
+        returnValues: {
+          donorId,
+          bundleId,
+          transferId
+        },
+        blockNumber: 1,
+        logIndex: 0
+      },
+      {
+        returnValues: {
+          donorId,
+          bundleId,
+          transferId: 100
+        },
+        blockNumber: 2,
+        logIndex: 0
+      },
+      {
+        returnValues: {
+          donorId,
+          bundleId,
+          transferId
+        },
+        blockNumber: 2,
+        logIndex: 1
+      }];
+
+    beforeEach(() => {
+      transfersRepository = new TransfersRepository();
+    });
+
+    it('extracts the blockNumber, logIndex, the fields specified in selector and appends event type', async () => {
+      expect(transfersRepository.prepareResolutionEvent(events, ['transferId', 'donorId', 'bundleId'])).to.deep.equal([
+        {donorId, bundleId, transferId, blockNumber: 1, logIndex: 0},
+        {donorId, bundleId, transferId: 100, blockNumber: 2, logIndex: 0},
+        {donorId, bundleId, transferId, blockNumber: 2, logIndex: 1}
+      ]);
+      expect(transfersRepository.prepareResolutionEvent(events, ['transferId'])).to.deep.equal([
+        {transferId, blockNumber: 1, logIndex: 0},
+        {transferId: 100, blockNumber: 2, logIndex: 0},
+        {transferId, blockNumber: 2, logIndex: 1}
+      ]);
+    });
+  });
+
+  describe('ongoingTransfers', () => {
+    const donorId = 1;
+    const bundleId = 2;
+    const transferId = 3;
+    // const fromBlock = 4;
```

Comment:
Comment should be removed

Reply:
This is a temporary solution - the code has changed and it is necessary to rewrite the test code.

Reply:
Fixed